kvbc/src/merkle_tree_db_adapter.cpp

```diff
@@ -325,14 +318,22 @@ Sliver DBAdapter::createBlockNode(const SetOfKeyValuePairs &updates, BlockId blo
   return block::detail::createNode(node);
 }
 
-Status DBAdapter::getBlockById(BlockId blockId, Sliver &block) const {
+RawBlock DBAdapter::getRawBlock(const BlockId &blockId) const {
   const auto blockKey = DBKeyManipulator::genBlockDbKey(blockId);
   Sliver blockNodeSliver;
-  if (auto status = db_->get(blockKey, blockNodeSliver); status.isNotFound()) {
+  if (auto statusNode = db_->get(blockKey, blockNodeSliver); statusNode.isNotFound()) {
     // If a block node is not found, look for a state transfer block.
-    return db_->get(DBKeyManipulator::generateSTTempBlockKey(blockId), block);
-  } else if (!status.isOK()) {
-    return status;
+    Sliver block;
+    if (auto statusSt = db_->get(DBKeyManipulator::generateSTTempBlockKey(blockId), block); statusSt.isNotFound()) {
```

Comment:
generateSTTempBlockKey() - is it a state transfer related function?

Reply:
It is not used by state transfer itself. Instead, it is used by the merkle tree DBAdapter to accumulate blocks as they arrive from state transfer.